pulsaradmin/pkg/admin/topic_test.go

```diff
@@ -213,3 +213,204 @@ func TestNonPartitionState(t *testing.T) {
+	assert.NoError(t, err)
+	err = admin.Topics().Create(*topicName, 3)
+	assert.NoError(t, err)
+	topicMetadata, err := admin.Topics().GetMetadata(*topicName)
+	assert.NoError(t, err)
+	assert.Equal(t, 3, topicMetadata.Partitions)
+
+	err = admin.Topics().Update(*topicName, 4)
+	assert.NoError(t, err)
+	topicMetadata, err = admin.Topics().GetMetadata(*topicName)
+	assert.NoError(t, err)
+	assert.Equal(t, 4, topicMetadata.Partitions)
+}
+
+func TestGetMessageID(t *testing.T) {
+	randomName := newTopicName()
+	topic := "persistent://public/default/" + randomName
+	topicPartitionZero := topic + "-partition-0"
+	cfg := &config.Config{}
+	admin, err := New(cfg)
+	assert.NoError(t, err)
+	assert.NotNil(t, admin)
+	topicName, err := utils.GetTopicName(topic)
+	topicPartitionZeroName, err := utils.GetTopicName(topicPartitionZero)
+	assert.NoError(t, err)
+	err = admin.Topics().Create(*topicName, 1)
+	assert.NoError(t, err)
+	ctx := context.Background()
+
+	// create consumer
+	client, err := pulsar.NewClient(pulsar.ClientOptions{
+		URL: lookupURL,
+	})
+	assert.NoError(t, err)
+	defer client.Close()
+	consumer, err := client.Subscribe(pulsar.ConsumerOptions{
+		Topic:            topic,
+		SubscriptionName: "my-sub",
+		Type:             pulsar.Exclusive,
+	})
+	assert.NoError(t, err)
+	defer consumer.Close()
+
+	// create producer
+	producer, err := client.CreateProducer(pulsar.ProducerOptions{
+		Topic:           topic,
+		DisableBatching: false,
+	})
+	assert.NoError(t, err)
+	defer producer.Close()
+	_, err = producer.Send(ctx, &pulsar.ProducerMessage{
+		Payload: []byte("hello"),
+		Key:     "pulsar",
+		Properties: map[string]string{
+			"key-1": "pulsar-1",
+		},
+	})
+	assert.NoError(t, err)
+
+	// ack message
+	msg, err := consumer.Receive(ctx)
+	assert.Equal(t, []byte("hello"), msg.Payload())
+	assert.Equal(t, "pulsar", msg.Key())
+	err = consumer.Ack(msg)
+	assert.NoError(t, err)
+
+	messasgeId, err := admin.Topics().GetMessageID(
+		*topicPartitionZeroName,
+		msg.PublishTime().Unix()*1000-1000,
+	)
+	assert.NoError(t, err)
+	assert.Equal(t, msg.ID().EntryID(), messasgeId.EntryID)
+	assert.Equal(t, msg.ID().LedgerID(), messasgeId.LedgerID)
+	assert.Equal(t, int(msg.ID().PartitionIdx()), messasgeId.PartitionIndex)
+}
+
+func TestMessageTTL(t *testing.T) {
+	randomName := newTopicName()
+	topic := "persistent://public/default/" + randomName
+	cfg := &config.Config{}
+	admin, err := New(cfg)
+	assert.NoError(t, err)
+	assert.NotNil(t, admin)
+	topicName, err := utils.GetTopicName(topic)
+	assert.NoError(t, err)
+	err = admin.Topics().Create(*topicName, 4)
+	assert.NoError(t, err)
+
+	messageTTL, err := admin.Topics().GetMessageTTL(*topicName)
+	assert.NoError(t, err)
+	assert.Equal(t, 0, messageTTL)
+	err = admin.Topics().SetMessageTTL(*topicName, 600)
+	assert.NoError(t, err)
+	//	topic policy is an async operation,
+	//	so we need to wait for a while to get current value
+	time.Sleep(10 * time.Second)
```

Comment:
You should avoid calling `Sleep` in tests. Please use something like `assert.Eventually`

Reply:
fixed